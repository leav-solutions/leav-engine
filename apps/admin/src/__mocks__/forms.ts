// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {GET_FORM_forms_list} from '../_gqlTypes/GET_FORM';
import {GET_FORMS_LIST_forms_list} from '../_gqlTypes/GET_FORMS_LIST';
import {FormElementTypes} from '../_gqlTypes/globalTypes';

const myField = {
    id: '123456',
    order: 0,
    uiElementType: 'field_container',
    containerId: 'some_container',
    settings: [
        {key: 'attribute', value: 'test_attribute'},
        {key: 'required', value: false},
        {key: 'input', value: 'text'}
    ],
    type: FormElementTypes.layout
};

export const mockFormLight: GET_FORMS_LIST_forms_list = {
    id: 'test_form_light',
    system: false,
    label: {fr: 'Test Form'}
};

export const mockFormFull: GET_FORM_forms_list = {
    ...mockFormLight,
    id: 'test_form_full',
    system: false,
    dependencyAttributes: [],
    label: {fr: 'Test Form'},
    elements: [{dependencyValue: null, elements: [{...myField}]}]
};
