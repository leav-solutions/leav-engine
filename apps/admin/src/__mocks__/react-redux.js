// Copyright LEAV Solutions 2017 until 2023/11/05, Copyright Aristid from 2023/11/06
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
module.exports = {
    connect: (mapStateToProps, mapDispatchToProps) => ReactComponent => ({
        mapStateToProps,
        mapDispatchToProps,
        ReactComponent
    }),
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
    Provider: ({children}) => children
};
