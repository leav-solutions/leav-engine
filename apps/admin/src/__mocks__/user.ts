// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {RecordIdentity, RecordIdentity_whoAmI_preview} from '_gqlTypes/RecordIdentity';

export const mockPreviews: RecordIdentity_whoAmI_preview = {
    tiny: '/fake/url/tiny.jpg',
    small: '/fake/url/small.jpg',
    medium: '/fake/url/medium.jpg',
    big: '/fake/url/big.jpg',
    huge: '/fake/url/huge.jpg',
    pdf: ''
};

export const mockModifier: RecordIdentity = {
    whoAmI: {
        id: '123456',
        label: 'admin',
        library: {
            id: 'users',
            label: {fr: 'Utilisateurs'}
        },
        preview: mockPreviews,
        color: '#123456'
    }
};
