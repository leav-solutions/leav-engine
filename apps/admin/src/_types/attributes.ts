// Copyright LEAV Solutions 2017 until 2023/11/05, Copyright Aristid from 2023/11/06
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {RecordIdentity_whoAmI} from '../_gqlTypes/RecordIdentity';

export interface ILinkValuesList {
    whoAmI: RecordIdentity_whoAmI;
}

export interface ITreeValuesListAncestors {
    record: {
        whoAmI: RecordIdentity_whoAmI;
    };
}

export interface ITreeValuesList {
    id: string;
    record: {
        whoAmI: RecordIdentity_whoAmI;
    };
    ancestors: ITreeValuesListAncestors[] | null;
}

export interface IValuesListConf {
    enable: boolean;
    allowFreeEntry?: boolean | null;
    allowListUpdate?: boolean | null;
    values?: ValuesList | null;
}

export type StandardValuesListType = Array<string | IDateRangeValue>;
export type ValuesList = StandardValuesListType | ILinkValuesList[] | ITreeValuesList[];

export interface IDateRangeValue {
    from: string;
    to: string;
}
