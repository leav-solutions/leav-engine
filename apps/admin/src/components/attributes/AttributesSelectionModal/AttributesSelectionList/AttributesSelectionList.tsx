// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import React from 'react';
import {useTranslation} from 'react-i18next';
import {Checkbox, Table} from 'semantic-ui-react';
import {GET_ATTRIBUTES_attributes_list} from '../../../../_gqlTypes/GET_ATTRIBUTES';

interface IAttributesSelectionListProps {
    attributes: GET_ATTRIBUTES_attributes_list[];
    selection: string[];
    toggleSelection: (attribute: GET_ATTRIBUTES_attributes_list) => void;
}

const AttributesSelectionList = (props: IAttributesSelectionListProps): JSX.Element => {
    const {t} = useTranslation();
    const {attributes, toggleSelection, selection} = props;

    return (
        <Table selectable striped>
            <Table.Header>
                <Table.Row>
                    <Table.HeaderCell />
                    <Table.HeaderCell>{t('attributes.label')}</Table.HeaderCell>
                    <Table.HeaderCell>{t('attributes.ID')}</Table.HeaderCell>
                </Table.Row>
            </Table.Header>
            <Table.Body>
                {attributes.map(a => {
                    const attrLabel = a.label !== null ? a.label.fr || a.label.en || a.id : a.id;
                    const onClick = () => toggleSelection(a);
                    const isSelected = selection.indexOf(a.id) !== -1;
                    return (
                        <Table.Row key={a.id} onClick={onClick} active={isSelected}>
                            <Table.Cell>
                                <Checkbox readOnly checked={isSelected} />
                            </Table.Cell>
                            <Table.Cell>{attrLabel}</Table.Cell>
                            <Table.Cell>{a.id}</Table.Cell>
                        </Table.Row>
                    );
                })}
            </Table.Body>
        </Table>
    );
};

export default AttributesSelectionList;
