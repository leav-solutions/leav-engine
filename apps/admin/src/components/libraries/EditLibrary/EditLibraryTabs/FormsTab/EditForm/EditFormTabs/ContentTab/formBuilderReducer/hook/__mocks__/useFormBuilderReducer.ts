// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import React, {Dispatch} from 'react';
import {FormBuilderAction, IFormBuilderState} from '../../formBuilderReducer';
import {mockInitialState} from '../../_fixtures/fixtures';

const initialState: IFormBuilderState = {
    form: null,
    library: '',
    openSettings: false,
    activeDependency: null,
    elementInSettings: null,
    elements: {},
    activeElements: {}
};

export const FormBuilderReducerContext = React.createContext<{
    state: IFormBuilderState;
    dispatch: Dispatch<FormBuilderAction>;
}>({state: mockInitialState, dispatch: () => jest.fn()});

export const useFormBuilderReducer = () => ({
    state: mockInitialState,
    dispatch: jest.fn()
});
