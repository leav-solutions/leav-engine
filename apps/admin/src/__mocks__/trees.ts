// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import pick from 'lodash/pick';
import {GET_TREE_BY_ID_trees_list} from '../_gqlTypes/GET_TREE_BY_ID';
import {PermissionsRelation, TreeBehavior} from '../_gqlTypes/globalTypes';
import {mockAttrTree} from './attributes';

export const mockTree: GET_TREE_BY_ID_trees_list = {
    id: 'test_tree',
    system: false,
    libraries: [
        {
            library: {id: 'test_lib', label: {fr: 'Test Lib'}, attributes: [{...mockAttrTree}]},
            settings: {allowMultiplePositions: false, allowedAtRoot: true, allowedChildren: ['__all__']}
        },
        {
            library: {id: 'test_lib2', label: {fr: 'Test Lib 2'}, attributes: [{...mockAttrTree}]},
            settings: {allowMultiplePositions: false, allowedAtRoot: true, allowedChildren: ['__all__']}
        }
    ],
    behavior: TreeBehavior.standard,
    label: {
        fr: 'TestTree',
        en: 'TestTree'
    },
    permissions_conf: null
};

export const mockTreeWithPermConf: GET_TREE_BY_ID_trees_list = {
    ...mockTree,
    id: 'test_tree_with_perm',
    permissions_conf: [
        {
            libraryId: 'test_lib',
            permissionsConf: {
                permissionTreeAttributes: [pick(mockAttrTree, ['id', 'label'])],
                relation: PermissionsRelation.and
            }
        },
        {
            libraryId: 'test_lib2',
            permissionsConf: {
                permissionTreeAttributes: [pick(mockAttrTree, ['id', 'label'])],
                relation: PermissionsRelation.and
            }
        }
    ]
};
