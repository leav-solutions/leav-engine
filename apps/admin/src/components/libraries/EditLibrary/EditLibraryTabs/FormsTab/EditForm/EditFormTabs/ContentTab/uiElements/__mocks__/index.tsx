// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import React from 'react';
import {FieldTypes, IUIElement, UIElementTypes} from '../../_types';

function FakeComp(): JSX.Element {
    return <div>Mock</div>;
}

const baseMock = {
    component: <FakeComp />,
    canDrop: jest.fn(),
    settings: []
};

export const layoutElements: {[type in UIElementTypes]: IUIElement} = {
    [UIElementTypes.FIELDS_CONTAINER]: {
        ...baseMock,
        type: UIElementTypes.FIELDS_CONTAINER
    },
    [UIElementTypes.DIVIDER]: {
        ...baseMock,
        type: UIElementTypes.DIVIDER
    },
    [UIElementTypes.TEXT_BLOCK]: {
        ...baseMock,
        type: UIElementTypes.TEXT_BLOCK
    },
    [UIElementTypes.TABS]: {
        ...baseMock,
        type: UIElementTypes.TABS
    }
};

export const formElements: {[type in FieldTypes]: IUIElement} = {
    [FieldTypes.TEXT_INPUT]: {
        ...baseMock,
        type: FieldTypes.TEXT_INPUT
    },
    [FieldTypes.DATE]: {
        ...baseMock,
        type: FieldTypes.DATE
    },
    [FieldTypes.CHECKBOX]: {
        ...baseMock,
        type: FieldTypes.CHECKBOX
    },
    [FieldTypes.ENCRYPTED]: {
        ...baseMock,
        type: FieldTypes.ENCRYPTED
    },
    [FieldTypes.DROPDOWN]: {
        ...baseMock,
        type: FieldTypes.DROPDOWN
    },
    [FieldTypes.LINK]: {
        ...baseMock,
        type: FieldTypes.LINK
    },
    [FieldTypes.TREE]: {
        ...baseMock,
        type: FieldTypes.TREE
    }
};
