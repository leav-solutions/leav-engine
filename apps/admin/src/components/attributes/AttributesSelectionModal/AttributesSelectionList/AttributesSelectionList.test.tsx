// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import React from 'react';
import {create} from 'react-test-renderer';
import {GET_ATTRIBUTES_attributes_list} from '../../../../_gqlTypes/GET_ATTRIBUTES';
import {mockAttrSimple} from '../../../../__mocks__/attributes';
import AttributesSelectionList from './AttributesSelectionList';

describe('AttributesSelectionList', () => {
    test('Snapshot test', async () => {
        const attributes: GET_ATTRIBUTES_attributes_list[] = [
            {
                ...mockAttrSimple,
                id: 'test_attr',
                label: {
                    fr: 'Test',
                    en: 'Test'
                }
            }
        ];

        const toggleSelection = jest.fn();

        const comp = create(
            <AttributesSelectionList attributes={attributes} selection={[]} toggleSelection={toggleSelection} />
        );

        expect(comp).toMatchSnapshot();
    });
});
