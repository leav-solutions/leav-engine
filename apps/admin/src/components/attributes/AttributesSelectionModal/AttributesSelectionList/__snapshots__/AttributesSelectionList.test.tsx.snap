// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributesSelectionList Snapshot test 1`] = `
<table
  className="ui selectable striped table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
      />
      <th
        className=""
      >
        attributes.label
      </th>
      <th
        className=""
      >
        attributes.ID
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className=""
      onClick={[Function]}
    >
      <td
        className=""
      >
        <div
          className="ui fitted read-only checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            readOnly={true}
            tabIndex={0}
            type="checkbox"
          />
          <label />
        </div>
      </td>
      <td
        className=""
      >
        Test
      </td>
      <td
        className=""
      >
        test_attr
      </td>
    </tr>
  </tbody>
</table>
`;
